File: sociallibrary/src/main/java/com/jy/sociallibrary/ext/data/StatusLiveData.java
package com.jy.sociallibrary.ext.data;

import androidx.lifecycle.MutableLiveData;

public class StatusLiveData extends MutableLiveData<StatusBean> {
    private StatusLiveData() {
    }

    private static class Holder {
        public static final StatusLiveData INSTANCE = new StatusLiveData();
    }

    public static StatusLiveData getInstance() {
        return Holder.INSTANCE;
    }


}

File: sociallibrary/src/main/java/com/jy/sociallibrary/ext/data/StatusLiveData.kt
package com.jy.sociallibrary.ext.data

import androidx.lifecycle.MutableLiveData

object StatusLiveData : MutableLiveData<StatusBean>() {

    fun getInstance(): StatusLiveData = Holder.INSTANCE

    private object Holder {
        val INSTANCE: StatusLiveData = StatusLiveData
    }
}